en/ros/external_position_estimation.md

```diff
@@ -1,114 +1,220 @@
+Reducing them will cause the estimator to trust the incoming pose estimate more. 
+You may need to set them lower than the allowed minimum and force-save. 
+
+> **Tip** If performance is still poor, try increasing the [LPE_PN_V](../advanced/parameter_reference.md#LPE_PN_V) parameter. 
+  This will cause the estimator to trust measurements more during velocity estimation.
+
+  
+## Working with ROS
+
+ROS is not *required* for supplying external pose information, but is highly recommended as it already comes with good integrations with VIO and MoCap systems.
+PX4 must already have been set up as above.
+
+### Getting pose data into ROS
+
+* Install the `vrpn_client_ros` package
+* You can get each rigid body pose on an individual topic by running
+  ```bash
+  roslaunch vrpn_client_ros sample.launch server:=<mocap machine ip>
+  ```
+
+If you named the rigidbody as `robot1`, you will get a topic like `/vrpn_client_node/robot1/pose`
+
+
+### Relaying Pose Data to PX4
+
+MAVROS has plugins to relay a visual estimation from a VIO or MoCap system using the following pipelines:
+
+ROS | MAVLink | uORB 
+--- | --- | ---
+/mavros/vision_pose/pose | [VISION_POSITION_ESTIMATE](https://mavlink.io/en/messages/common.html#VISION_POSITION_ESTIMATE) | `vehicle_visual_odometry`
+/mavros/odometry/odom | [ODOMETRY](https://mavlink.io/en/messages/common.html#ODOMETRY) (`frame_id =` [MAV_FRAME_VISION_NED](https://mavlink.io/en/messages/common.html#MAV_FRAME_VISION_NED)) | `vehicle_visual_odometry`
+/mavros/mocap/pose | [ATT_POS_MOCAP](https://mavlink.io/en/messages/common.html#ATT_POS_MOCAP) | `vehicle_mocap_odometry`
+/mavros/odometry/odom | [ODOMETRY](https://mavlink.io/en/messages/common.html#ODOMETRY) (`frame_id =` [MAV_FRAME_MOCAP_NED](https://mavlink.io/en/messages/common.html#MAV_FRAME_MOCAP_NED)) | `vehicle_mocap_odometry`
+
+You can use any of the above pipelines with LPE. 
 
-> **Tip**If performance is still poor, try increasing the `LPE_PN_V` parameter. This will cause the estimator to trust measurements more during velocity estimation..
+If you're working with EKF2, only the "vision" pipelines are supported. 
+To use MoCap data with EKF2 you will have to [remap](http://wiki.ros.org/roslaunch/XML/remap) the pose topic that you get from MoCap:
+- MoCap ROS topics of type `geometry_msgs/PoseStamped` or `geometry_msgs/PoseWithCovarianceStamped` must be remapped to `/mavros/vision_pose/pose`. 
+  The `geometry_msgs/PoseStamped` topic is most common as MoCap doesn't usually have associated covariances to the data.
+- If you get data through a `nav_msgs/Odometry` ROS message then you will need to remap it to `/mavros/odometry/odom`.
+- TBD from `/vrpn_client_node/<rigid_body_name>/pose` to `/mavros/vision_pose/pose`. 
 
-## Asserting on Reference Frames
 
-This section shows how to setup the system with the proper reference frames. There are various representations but we will use two of them: ENU and NED. 
+## Asserting on Reference Frames 
```

Comment:
This section confuses me - what does "Asserting on Reference Frames" even mean? :-)
The rest of the section and sub headings talk about "rigid body" which is not defined, and is all about Optitrack. 

So:
1. how relevant is this reference frame information/how generic?
2. If it is relevant how do we separate it. I'm "guessing" that it probably is useful because maybe ROS uses ENU and PX4 uses NED, so this could be under ROS heading with a note to that effect?

The mocap specific info could then go under the specific optitrack setup?

Reply:
The frame of the rigid body is the body frame. If it is relevant as a rigid body, probably not.

It's relevant to state that the local/world frames are different from ROS to PX4:
- ROS body frame is FLU (X forward, Y left, Z up), usually named `base_link`, while in PX4 is FRD (X forward, Y right and Z down);
- ROS world frame is in ENU (X pointing East, Y pointing North and Z pointing up), with the naming being `odom` or `map`, while in PX4, the frame is NED (X North, Y East, Z down).

For a better insight about the ROS frames: http://www.ros.org/reps/rep-0105.html

Reply:
Thanks @TSC21 . I think I have captured what is relevant in https://github.com/PX4/Devguide/pull/681/commits/259d327289374a9426e22720e693d40a7dfcba4a

The "note" in that section on MoCap may not be needed.